### PR TITLE
refactor(localdebug): move post localdebug into solution

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Detail: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/12863188

### CODE REVIEW

File: packages/fx-core/src/common/localEnvMultiProvider.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import * as dotenv from "dotenv";
 import * as fs from "fs-extra";
```
Comment (swatDong) [inline packages/fx-core/src/common/localEnvMultiProvider.ts:6]: Just call it `LocalEnvProvider`.
Move it to *common/local/*

File: packages/fx-core/src/common/constants.ts [hunk 2]
```diff
@@ -61,3 +61,9 @@ export class FeatureFlagName {
   static readonly rootDirectory = "TEAMSFX_ROOT_DIRECTORY";
   static readonly VSCallingCLI = "VS_CALLING_CLI";
 }
+
+export class FolderName {
```
Comment (swatDong) [inline packages/fx-core/src/common/constants.ts:65]: Move to *common/local/*

File: packages/fx-core/src/common/localEnvMultiProvider.ts (hunk 1 above)
Comment (XiaofuHuang) [reply packages/fx-core/src/common/localEnvMultiProvider.ts]: Got it.

**Review by kuojianlu — APPROVED** [review]

File: packages/fx-core/tests/plugins/solution/debug/util/certificate.test.ts [hunk 3]
```diff
@@ -57,7 +79,7 @@ describe("certificate", () => {
       chai.assert.isTrue(fs.pathExistsSync(expectedKeyFile));
       const keyContent = fs.readFileSync(expectedKeyFile, { encoding: "utf8" });
       chai.assert.isTrue(
-        /-----BEGIN RSA PRIVATE KEY-----.*-----END RSA PRIVATE KEY-----/gs.test(keyContent)
+        /-----BEGIN RSA .RIVATE KEY-----.*-----END RSA .RIVATE KEY-----/gs.test(keyContent)
```
Comment (swatDong) [inline packages/fx-core/tests/plugins/solution/debug/util/certificate.test.ts:82]: Why this change?
We do need to ensure the cert content. If cred scan does not pass, add to */.github/detect/excludes.txt*.

File: packages/fx-core/src/common/localEnvMultiProvider.ts (hunk 1 above)
Comment (swatDong) [reply packages/fx-core/src/common/localEnvMultiProvider.ts]: Update class name as well.

File: packages/fx-core/tests/plugins/solution/debug/util/certificate.test.ts (hunk 3 above)
Comment (XiaofuHuang) [reply packages/fx-core/tests/plugins/solution/debug/util/certificate.test.ts]: Thanks! Put it into the exclude list and passed the check.

**Review by swatDong — APPROVED** [review]

**Review by 1yefuwang1 — DISMISSED** [review]

File: packages/fx-core/src/plugins/solution/fx-solution/debug/provisionLocal.ts [hunk 4]
```diff
@@ -129,3 +140,136 @@ export async function setupLocalDebugSettings(
...
+    "trust-development-certificate": trustDevCert + "",
+  };
+  TelemetryUtils.init(ctx);
+  TelemetryUtils.sendStartEvent(TelemetryEventName.configLocalDebugSettings, telemetryProperties);
+
+  try {
+    if (inputs.platform === Platform.VSCode || inputs.platform === Platform.CLI) {
+      const isMigrateFromV1 = ContextHelper.isMigrateFromV1(ctx);
+
+      const localEnvMultiProvider = new LocalEnvProvider(inputs.projectPath!);
```
Comment (dooriya) [inline packages/fx-core/src/plugins/solution/fx-solution/debug/provisionLocal.ts:170]:
```Suggestion
      const localEnvProvider = new LocalEnvProvider(inputs.projectPath!);
```
Since you're already renamed the type  name, also rename the variable consistently.

Comment (XiaofuHuang) [reply packages/fx-core/src/plugins/solution/fx-solution/debug/provisionLocal.ts]: Thanks! Renamed them

**Review by dooriya — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Review by 1yefuwang1 — APPROVED** [review]
